api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -381,47 +415,73 @@ private void validateAreaOfResearch(Profile profile) throws BadRequestException
+   * @param previousProfile
    * @throws BadRequestException
    */
   @VisibleForTesting
-  public void validateProfile(@Nonnull Profile updatedProfile, @Nullable Profile prevProfile)
+  public void validateProfile(
+      @Nonnull Profile updatedProfile, @Nullable Profile previousProfile, boolean userIsAdmin) {
+
+    final boolean isNewProfile = previousProfile == null;
+    final Diff diff = javers.compare(previousProfile, updatedProfile);
+
+    validateProfileForCorrectness(updatedProfile, isNewProfile, diff);
+
+    if (userIsAdmin) {
+      validateChangesAllowedByAdmin(diff);
+    } else {
+      validateChangesAllowedByUser(diff);
+    }
+  }
+
+  private void validateProfileForCorrectness(Profile profile, boolean isNewProfile, Diff diff)
       throws BadRequestException {
-    boolean isNewObject = prevProfile == null;
-    Diff diff = javers.compare(prevProfile, updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "username")) {
+      validateUsername(profile);
+    }
+    if (isNewProfile || fieldChanged(diff, "contactEmail")) {
+      validateContactEmail(profile);
 
-    if (!getChangesWithPrefix(diff, "username").isEmpty() || isNewObject) {
-      validateUsername(updatedProfile);
+      // only validate if the new profile has an affiliation - some older users do not
+      if (profile.getVerifiedInstitutionalAffiliation() != null) {
+        validateInstitutionalAffiliationAgainstEmail(profile);
+      }
     }
-    if (!getChangesWithPrefix(diff, "contactEmail").isEmpty() || isNewObject) {
-      validateContactEmail(updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "givenName")) {
+      validateGivenName(profile);
     }
-    if (!getChangesWithPrefix(diff, "givenName").isEmpty() || isNewObject) {
-      validateGivenName(updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "familyName")) {
+      validateFamilyName(profile);
     }
-    if (!getChangesWithPrefix(diff, "familyName").isEmpty() || isNewObject) {
-      validateFamilyName(updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "address")) {
+      validateAddress(profile);
     }
-    if (!getChangesWithPrefix(diff, "address").isEmpty() || isNewObject) {
-      validateAddress(updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "areaOfResearch")) {
+      validateAreaOfResearch(profile);
     }
-    if (!getChangesWithPrefix(diff, "areaOfResearch").isEmpty() || isNewObject) {
-      validateAreaOfResearch(updatedProfile);
+    if (isNewProfile || fieldChanged(diff, "verifiedInstitutionalAffiliation")) {
+      validateInstitutionalAffiliation(profile);
     }
-    if (!getChangesWithPrefix(diff, "verifiedInstitutionalAffiliation").isEmpty() || isNewObject) {
-      validateInstitutionalAffiliation(updatedProfile);
+  }
+
+  private void validateChangesAllowedByUser(Diff diff) {
+    if (fieldChanged(diff, "username")) {
+      // See RW-1488.
+      throw new BadRequestException("Changing username is not supported");
```

Comment:
By exiting early, we can't validate multiple fields and give the client a single exception with a list of issues with the profile. Granted no one has asked for that yet.

Reply:
Good point!  I filed this: https://precisionmedicineinitiative.atlassian.net/browse/RW-5334